### DISCUSSION TITLE
Is It Possible To Simulate Orbits For Individual Satellite Systems Using GROOPS?

### DISCUSSION THREAD

**Discussion by chilk-gnss** [groops-devs/groops#85] [Q&A] [answered]
Hi GROOPS Developers,

I am interested in simulating the orbits of different satellite systems. I followed the instructions given in the documentation especially the [SimulateOrbit.](https://groops-devs.github.io/groops/html/SimulateOrbit.html). Taking an example of MEO satellites, I have the following plot:

**=> MEO GroundTrack: 2021-11-01**
![groundTrack_MEO_2021-11-01](https://user-images.githubusercontent.com/109251614/182130340-047dade2-053d-4f71-9626-42addf2b1e76.png)

However, I wanted all the MEO satellites alongside the other satellites to appear in the plot, for example for BEIDOU. Here is an example for IGSO:

**=> IGSO GroundTrack: 2021-11-01**
![groundTrack_IGSO_2021-11-01](https://user-images.githubusercontent.com/109251614/182129666-1d8c39f8-c1d5-482b-b549-3f658efcb0e5.png)

Question: How can I get MEO, IGSO, and GEO satellite tracks in the same plot? 
Goal: To have all the MEO, IGSO, and GEO satellites in the same plot.  

You may refer to my script attached

**=> multiSimulateOrbit.xml**
[multiSimulateOrbit.zip](https://github.com/groops-devs/groops/files/9233192/multiSimulateOrbit.zip)

Thanks for your support.

Regards,

Chilk.

**Comment by tmayerguerr** [comment]
You can add multiple layers in PlotMap (for each satellite: layer->points with the respective inputfileGriddedData). You must call Orbit2Groundtracks for each satellite before (and of course  with different outputfileGriddedData). 
regards
Torsten

**Comment by chilk-gnss** [comment]
Dear Professor,

Thanks for your timely response.

I also noted that the PlotMap can accept multiple layers. However, the [**SimulateOrbit**](https://groops-devs.github.io/groops/html/SimulateOrbit.html) cannot accommodate multiple layers in terms of **integrationConstants**. If the settings for the **propagator**, **earthRotation**, **ephemerides**, and **forces** are to be kept unchanged in the **SimulateOrbit**, then it will be hard to prepare the output orbit files (including **inputfileGriddedData**) for multiple constellations. 

Another issue, when I try to plot the GEO satellites with zero inclination angle, I obtain an empty figure. Further, I am not sure how to obtain the tracks for all satellites so that I can proceed with my experiments on GRACE. I was, therefore, wondering if you could provide an example on how I can make such plots.

Thanks for your understanding and support.

Kind regards,

Chilk.

**Comment by tmayerguerr** [comment]
You can add the program [SimulateOrbit](https://groops-devs.github.io/groops/html/SimulateOrbit.html) itself multiple times, each with different integration constants.
A GEO satellite with zero inclination angle is a geostationary satellite. Therefore the groundtrack is a single constant point at Earth.

**Comment by Waley04** [comment]
Hi, 

If I understand @chilk-gnss  well, he is looking for individual satellite tracks for a given constellation. From what I have noted, the integration constants in the [SimulateOrbit](https://groops-devs.github.io/groops/html/SimulateOrbit.html) may be ideal for generating the representative ground tracks of the GNSS constellation. 

Following your advice, I can generate representative tracks for any satellite system. However, I do not know how to get the tracks for each satellite for a specific day in this regard. For example, generating the tracks for 32 GPS PRNs. 

Thus, your guidance will assist us.

Warm regards,

Waley.

**Comment by tmayerguerr** [comment, marked as answer]
Hmm, still quite unsure that the question is. If you have integration constants of each satellite at hand you can you SimulateOrbit. Much easier is to download an SP3 file with all the real orbits from IGS/MEGX. Attached you find an example for plotting all orbits from WHU MGEX solution for a single day:
[groundtracks.zip](https://github.com/groops-devs/groops/files/9242958/groundtracks.zip).

regards
Torsten

**Comment by chilk-gnss** [comment]
Thanks very much Professor. 

Using your sample script, I have managed to solve what I was looking for alongside [SimulateStarCamera](https://groops-devs.github.io/groops/html/SimulateStarCamera.html). 

Best regards,

Chilk.

**Comment by Waley04** [comment]
It has also worked for me. I can now extend it further to solar radiation pressure modelling.

**Comment by gnssLa** [comment]
Prof. @tmayerguerr,

For the sake of curiosity, how can I plot the number of satellites on a map using color, and show the variation in the number of satellites on a color bar?  

Like this =>
![image](https://github.com/groops-devs/groops/assets/109293325/2976e043-9b44-4d19-8e7b-7bcc6e4dc680)

Thanks for your help. 

Regards,

Duarte.
